### PR TITLE
Feat: preserve unique onnx node names

### PR DESCRIPTION
**Pull request by jan-haug**
Currently, each ONNX node is assigned a unique name consisting of the domain, op_type and a running counter to name the torch module with: https://github.com/ENOT-AutoDL/onnx2torch/blob/main/onnx2torch/onnx_graph.py#L29-L34

However, it can be useful to be able to infer the torch module from the onnx graph after loading it, say for example if you want to modify something specifically in the conv-layers in the head of the network. Since torch 1.13, the names of the onnx nodes are also inferred from the namescopes (see https://github.com/pytorch/pytorch/issues/75100), so being able to match that makes finding modules after a torch-onnx-torch round extremely easy.

This change replaces the previous naming scheme with one based on the onnx node names, modifying them only when and as far as necessary:
- Dots `.` are not allowed as names, so they are replaced with a slash `/`. They occur when you export e.g. a `ModuleList` or `Sequential` from torch. Colons `:` are fine and don't need handling
- If for some reason, there are duplicate node names, a running counter is appended with an underscore. This keeps the existing counter, but doesn't append anything on the first occurence to keep the name if possible. Generally, node names are required to be unique in the ONNX standard anyways, but nothing prevents you from creating a graph with duplicate names, so why not account for it anyways.
- If for some reason, the name is empty, it will use the op_type as a fallback, so basically the old behavior
- `domain` is prepended and separated by a slash, too. Not sure if that is needed, in my use-cases `domain` is empty anyways

Example screenshots from inspecting the modules after loading a cifar100 resnet:
<details>
<summary>Before:</summary>

</details>
<details>
<summary>After:</summary>

</details>

Also, I'm assuming the existing tests are enough to make sure this doesn't actually break things. Should I add a test to make sure it actually works as expected?

### CODE REVIEW

**Comment by senysenyseny16** [comment]
Hi, @jan-haug.

Thanks for your PR.
New naming scheme looks nice!

Could you create separate function for node name generation?

Something like this:
```python3
@staticmethod
def generate_name(node: NodeProto) -> str:
    ...
```

**Comment by jan-haug** [comment]
@senysenyseny16 Thank you!
Like this? https://github.com/ENOT-AutoDL/onnx2torch/pull/140/commits/b2c770ac9bc6576dd3c994b2644fb64c80b0a1ea

File: onnx2torch/onnx_graph.py [hunk 1]
```diff
@@ -94,3 +94,7 @@ def value_as_node_output(  # pylint: disable=missing-function-docstring
         value_name: str,
     ) -> Tuple[OnnxNode, int]:
         return self._node_output_values[value_name]
+
+    @staticmethod
+    def generate_node_name(node: NodeProto) -> str:
+        return (f'{node.domain}/' + (node.name.replace(".", "/") or node.op_type)).lstrip("/")
```
Comment (senysenyseny16) [inline onnx2torch/onnx_graph.py:100]:
```suggestion
        return (f'{node.domain}/' + (node.name.replace('.', '/') or node.op_type)).lstrip('/')
```

**Review by senysenyseny16 — CHANGES_REQUESTED** [review]

File: onnx2torch/onnx_graph.py (hunk 1 above)
Comment (senysenyseny16) [reply onnx2torch/onnx_graph.py:100]: Could you fix imports style?

Comment (jan-haug) [reply onnx2torch/onnx_graph.py]: Good catch! I added that :)

**Review by ivkalgin — APPROVED** [review]

**Comment by senysenyseny16** [comment]
@jan-haug Thank you!

**Pull request merged**